plugins/inventory/proxmox.py

```diff
@@ -0,0 +1,306 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) 2016 Guido Günther <agx@sigxcpu.org>, Daniel Lobato Garcia <dlobatog@redhat.com>
+# Copyright (c) 2018 Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = '''
+    name: proxmox
+    plugin_type: inventory
+    short_description: proxmox inventory source
+    version_added: "2.7"
+    requirements:
+        - requests >= 1.1
+    description:
+        - Get inventory hosts from the proxmox service.
+        - "Uses a configuration file as an inventory source, it must end in ``.proxmox.yml`` or ``.proxmox.yaml`` and has a ``plugin: proxmox`` entry."
+        - Will retrieve the first network interface with an IP for Proxmox nodes
+        - Can retrieve LXC/QEMU configuration as facts
+    extends_documentation_fragment:
+        - inventory_cache
+    options:
+      plugin:
+        description: the name of this plugin, it should alwys be set to 'proxmox' for this plugin to recognize it as it's own.
+        required: True
+        choices: ['proxmox']
+      url:
+        description: url to proxmox
+        default: 'http://localhost:8006'
+      user:
+        description: proxmox authentication user
+        required: True
+      password:
+        description: proxmox authentication password
+        required: True
+      validate_certs:
+        description: verify SSL certificate if using https
+        type: boolean
+        default: False
+      group_prefix:
+        description: prefix to apply to proxmox groups
+        default: proxmox_
+      facts_prefix:
+        description: prefix to apply to vm config facts
+        default: proxmox_
+      want_facts:
+        description: gather vm configuration facts
+        default: False
+'''
+
+EXAMPLES = '''
+# my.proxmox.yml
+plugin: proxmox
+url: http://localhost:8006
+user: ansible-tester
+password: secure
+validate_certs: False
+'''
+
+import re
+
+from ansible.module_utils.common._collections_compat import MutableMapping
+from distutils.version import LooseVersion
+
+from ansible.errors import AnsibleError
+from ansible.plugins.inventory import BaseInventoryPlugin, Cacheable
+from ansible.module_utils.six.moves.urllib.parse import urlencode
+
+
+# 3rd party imports
+try:
+    import requests
+    if LooseVersion(requests.__version__) < LooseVersion('1.1.0'):
+        raise ImportError
+except ImportError:
+    raise AnsibleError('This script requires python-requests 1.1 as a minimum version')
```

Comment:
Use a boolean flag for whether the import succeeded, and fail in the plugin instead.